components/raftstore/src/store/worker/compact.rs

```diff
@@ -127,6 +130,31 @@ where
         Runner { engine }
     }
 
+    /// Periodic full compaction.
+    ///
+    /// NOTE this is a highly experimental feature!
+    ///
+    /// TODO: Do not start if there is heavy I/O.
+    /// TODO: Make it possible to rate limit, pause, or abort this by compacting
+    /// a range at a time.
+    pub fn full_compact(&mut self) -> Result<(), Error> {
+        fail_point!("on_full_compact");
+        info!("full compaction started");
+        let timer = Instant::now();
+        let full_compact_timer = FULL_COMPACT.start_coarse_timer();
+        box_try!(self.engine.compact_range(
+            None, None, // Compact the entire key range.
+            true, // exclusive manual: do not run if background compaction is running
```

Comment:
This comment contradict RocksDB's doc https://github.com/facebook/rocksdb/wiki/Manual-Compaction#compactrange

Reply:
the comment says: ```CompactRangeOptions::exclusive_manual_compaction When set to true, no other compaction will run when this manual compaction is running. Default value is true```

i guess i should rephrase it: when this runs, no other compaction will run.